libpod/runtime_img.go

```diff
@@ -124,7 +126,74 @@ func (r *Runtime) RemoveImage(image *image.Image, force bool) (string, error) {
 		return "", fmt.Errorf("unable to delete %s (must force) - image is referred to in multiple tags", image.ID())
 	}
 
-	return image.ID(), image.Remove(force)
+	err = image.Remove(force)
+	if err != nil && errors.Cause(err) == storage.ErrImageUsedByContainer {
```

Comment:
I don't think we need this. We can just get storageContainers for the image and return an error if containers exist or remove them if --force is set. We don't need to do it in an error handling block, that makes things a lot harder to follow.

Reply:
I'm not sure exactly what you're suggesting here.  After line 129 are you saying I should just try getting the storage containers and then error if the image has containers and no force, other wise remove?  When we get to this point we've already tried removing the image and that failed due to it being a storage image and a container was associated.  I might be able to move the storage image handling that I've below into image.Remove() if that would be preferable.  I'll play with that and hit me up on IRC in the morn if I'm off base.

Reply:
Sorry, should have been more clear - I mean before 129, we can do the check if the container is mounted before the actual remove happens, thus removing the need to check errors.Cause() and perform complicated handling. This might make our fastpath a bit slower (we do a check unconditionally instead of conditionally), but will improve code flow and readability. We can always go back and make things faster later if need be.

Reply:
This chunk has been reworked and greatly reduced.  I still have the error check, but I think it makes more sense now.  Cand/will adjust after the next review if you think appropriate.